server/server.go

```diff
@@ -114,7 +103,29 @@ func (s *Server) serve(listener net.Listener, readyCb func(addr string)) error {
 		readyCb(l.Addr().String())
 	}
 
-	return s.httpServer.Serve(l)
+	for {
```

Comment:
Instead of using http.Server, we just handle connections ourselves and pass them off to the proxy package, which does most of the work.